src/Codeception/Application.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Codeception;
+
+use Codeception\Exception\ConfigurationException;
+use Symfony\Component\Console\Application as BaseApplication;
+use Symfony\Component\Console\Output\ConsoleOutput;
+use Codeception\Lib\Interfaces\CustomCommands;
+
+class Application extends BaseApplication
+{
+    /**
+     * Register commands from a config file
+     *
+     *  extensions:
+     *      commands:
+     *          - project\MyNewCommand
+     *
+     */
+    public function registerCustomCommands()
+    {
+        $aCommands = Configuration::config();
+
+        if (empty($aCommands['extensions']['commands'])) {
+            return;
+        }
+
+        try {
+            foreach ($aCommands['extensions']['commands'] as $sCommandClass) {
+                $sCommandName = $this->_getCommandName($sCommandClass);
+                $this->add(new $sCommandClass($sCommandName));
+            }
+        } catch (\Exception $e) {
+            $this->renderException($e, new ConsoleOutput());
+            exit;
```

Comment:
You should only exit if autoExit is enabled - BaseApplication::isAutoExitEnabled()